tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/WebRequest/HttpWebRequest_EndGetResponse_Integration.cs

```diff
@@ -94,9 +94,23 @@ public class HttpWebRequest_EndGetResponse_Integration
                         if (returnValue is HttpWebResponse response)
                         {
                             scope.Span.SetHttpStatusCode((int)response.StatusCode, isServer: false, Tracer.Instance.Settings);
+                            scope.Dispose();
                         }
+                        else if (exception is WebException { Status: WebExceptionStatus.ProtocolError, Response: HttpWebResponse exceptionResponse })
+                        {
+                            scope.Span.SetHttpStatusCode((int)exceptionResponse.StatusCode, isServer: false, Tracer.Instance.Settings);
 
-                        scope.DisposeWithException(exception);
+                            // SetHttpStatusCode will mark the span with an error if the StatusCode is within the configured range
+                            // Add the remaining error attributes
+                            scope.Span.SetTag(Trace.Tags.ErrorMsg, exception.Message);
```

Comment:
To avoid duplicating the code that sets the 3 error tags, what do you think of adding an optional parameter to `Span.SetException()`:

```csharp
void SetException(Exception exception, bool isError = true)
{
    // Error = true;
    Error = isError;

    // ...
}
```

and calling it here with `setErrorField: false`:
```csharp
else if (exception is...)
{
    scope.Span.SetHttpStatusCode(...);
    scope.Span.SetException(exception, setErrorField: false);
    scope.Dispose();
}
```

The idea

Reply:
Yeah I think that's a good idea, let me see about implementing it

Reply:
I added an update in 1cdd13915202be227e3f3d9540fba345457e1868 that adds a new Span.SetExceptionTags which I think should help with this. WDYT?